wgpu/src/lib.rs

```diff
@@ -4739,7 +4765,42 @@ impl<'a> Drop for RenderPass<'a> {
     }
 }
 
-impl ComputePass {
+impl<'a> ComputePass<'a> {
+    /// Drops lifetime constraint to the parent command encoder, making usage of the encoder
+    /// while this pass is recorded a runtime error instead.
+    ///
+    /// Attention: As long as the compute pass has not been ended, any mutable operation on the parent
+    /// command encoder will cause a runtime error and invalidate it!
+    /// By default the lifetime constraint protects from this, but it can be useful
+    /// to handle this at runtime, for example in order to store the pass and encoder in the same
+    /// data structure.
+    ///
+    /// This operation has no effect on pass recording and is considered low overhead.
+    /// It's a safe operation, since [`CommandEncoder`] is in a locked state as long as the pass is active
+    /// regardless of here lifted lifetime constraint.
+    pub fn make_static(mut self) -> ComputePass<'static> {
```

Comment:
I think the name `make_static()` needs improvement. When looking here at the types it's obvious what it means, but out of context when reading code, it could be misleading. “static” doesn't just mean `'static`; it can imply unchanging, compile-time, or leaked. Also, the way it is _used_ is essentially making the program “more dynamic(ally checked)”.

Separately, “`make_`” isn't a usual prefix for this sort of operation; [the convention](https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv) would be “`into_`”. In the standard library, `make_` is consistently used for `&mut self` operations that perform some mutation (and may then return another `&mut` obeying some constraint that has been applied).

How about calling this `without_lifetime()`? (The role `'static` is playing here is “no lifetime constraint” more than “something in here lives forever”.)

Or in a different direction, something like  `into_dynamically_checked()`?

Reply:
Makes a lot of sense to me, I agree that we need something better.

`into_dynamically_checked` goes into the right direction imho since it is a "into" operation at its heart just as you say. Only it doesn't quite convey what is checked and it makes it sound like it adds overhead when it doesn't 🤔 

What do you think about `into_encoder_independent_lifetime`? Still not amazing, but it captures that it changes the lifetime associating to the parent encoder 🤔

Reply:
> What do you think about `into_encoder_independent_lifetime`? Still not amazing, but it captures that it changes the lifetime associating to the parent encoder 🤔

“encoder independent lifetime” sounds like it has a lifetime that still means something specific, rather than one that is now insignificant. I really think that the name should make it clear that it's *removing* something, instead (that's why I suggested “`without_lifetime`”). The fact that there is now a `'static` lifetime is a necessary element of the implementation, but not significant itself. Analogy: Suppose we had a generic type `Foo<T>` and some way to type-erase it to `Foo<()>`. We wouldn’t call it something like “`into_unit_type_parameter()`”, but “`into_erased()`”, because the `()` is just filling the slot, not meaning something itself.

More brainstorming: 

* `.untie_lifetime()` (some people are starting to use "tied" as a description of a lifetime that appears in a function signature in a way that it actually links an input reference to an output reference, rather than, say, being only an input)
* `.forget_lifetime()`

Reply:
compelling case against `into_encoder_independent_lifetime`.
`untie_lifetime` makes sense to me as well, but I fear this has a bit too much unusual jargon.
`forget_lifetime` goes a little bit too much into `std::mem::forget`, but otherwise that's my favorite now, happy to leave it there